recipes/install.rb

```diff
@@ -10,60 +10,69 @@
 group = node[:bfd][:group]
 source_code_location = "#{Chef::Config[:file_cache_path]}/bfd"
 
-%w(ruby-dev ruby automake autoconf gcc g++ git make).each do |pkg|
+%w(checkinstall fakeroot automake autoconf gcc g++ git make).each do |pkg|
   package pkg do
     action :upgrade
   end
 end
 
-gem_package 'fpm' do
-  gem_binary '/usr/bin/gem'
-  action :install
-end
-
 git source_code_location do
    repository node[:bfd][:repo][:url]
    revision node[:bfd][:repo][:sha]
    action :sync
-   notifies :run, 'bash[buildtools_bfd]', :immediately
    not_if { ::File.exist?(target_filepath) }
 end
 
 bash 'buildtools_bfd' do
   cwd source_code_location
   code './autogen.sh'
-  action :nothing
-  notifies :run, 'bash[configure_bfd]', :immediately
+  action :run
+  not_if { ::File.exist?(target_filepath) }
 end
 
 bash 'configure_bfd' do
   cwd source_code_location
-  code "./configure --prefix=#{node[:bfd][:bin_dir]}/bfd-build"
-  action :nothing
-  notifies :run, 'bash[make_bfd]', :immediately
+  code './configure --prefix=/usr/local'
+  action :run
+  not_if { ::File.exist?(target_filepath) }
 end
 
 bash 'make_bfd' do
   cwd source_code_location
-  code 'make && make install'
-  action :nothing
-  notifies :run, 'bash[build_bfd_package]', :immediately
+  code 'make'
+  action :run
+  not_if { ::File.exist?(target_filepath) }
 end
 
 bash 'build_bfd_package' do
-  dependencies_str = node[:bfd][:package][:dependencies] ? "-d #{node[:bfd][:package][:dependencies]} --no-auto-depends" : "--no-depends --no-auto-depends"
-  cwd "#{node[:bfd][:bin_dir]}/bfd-build"
+  dependencies_str =
+    if node[:bfd][:package][:dependencies]
+      "--requires #{node[:bfd][:package][:dependencies]}"
+    else
+      ""
+    end
+  cwd source_code_location
   user 'root'
   group 'root'
-  code %Q{
-    fpm -s dir -t deb --prefix /usr/local \
-        -n #{node[:bfd][:package][:short_name]} \
-        -v #{node[:bfd][:version]} \
-        #{dependencies_str} \
-        * && \
-    mv #{node[:bfd][:package][:name]} #{target_filepath}
-  }
+  code 'checkinstall -D ' \
+    "  --pkgname #{node[:bfd][:package][:short_name]} " \
+    "  --pkgversion #{node[:bfd][:version]}" \
+    "  --pkgrelease #{node[:bfd][:release]}" \
+    "  -A #{node[:bfd][:arch]}"
   umask 0002
   not_if { ::File.exist?(target_filepath) }
-  action :nothing
+  action :run
+end
+
+bash 'copy_bfd_package' do
+  cwd source_code_location
+  code "/usr/bin/install -m 444 " \
+    "#{node[:bfd][:package][:name]} #{target_filepath}"
+  not_if { ::File.exist?(target_filepath) }
+  action :run
+end
+
+dpkg_package 'openbfdd' do
```

Comment:
This recipe might be poorly named and perhaps should be `build.rb` instead? But we have `default.rb` which also installs the dpkg. Can we have only one recipe to install the package?